src/airflow/dags/common_airflow.py

```diff
@@ -200,3 +220,20 @@ def delete_cluster(cluster_name: str) -> DataprocDeleteClusterOperator:
         trigger_rule=TriggerRule.ALL_DONE,
         deferrable=True,
     )
+
+
+def read_yaml_config(config_path):
+    """Parse a YAMl config file and do all necessary checks."""
+    assert config_path.exists(), f"YAML config path {config_path} does not exist."
+    with open(config_path, "r") as config_file:
+        return yaml.safe_load(config_file)
+
+
+def generate_dag(cluster_name, tasks):
+    """For a list of tasks, generate a complete DAG."""
+    return (
+        create_cluster(cluster_name)
```

Comment:
I am now trying to run LD index and have to tweak the cluster parameters.
I think this function could live in a more visible space, I expect that in common we have rather "static" logic.

Reply:
The idea is that a single cluster, under a uniform configuration, will be able to support all kinds of jobs. Very shortly I'll submit another PR which enables autoscaling.

I was wondering if you could please list the tweaks you are applying to the cluster config to run LD index? Then I can make sure that the unified configuration supports that as well :upside_down_face:

Reply:
This is what i am trying:
```
create_cluster(
                CLUSTER_NAME,
                master_machine_type="n1-highmem-96",
                master_disk_size=5000,
                num_local_ssds=0,
            )
```

Reply:
I generated a LD index yesterday based on these params.
When you try this with autoscaling, can you please change the output path to make sure we don't clash? I need this data for subsequent steps.
Also I'll shortly be submitting a PR that integrates this step as part of the preprocessing part.

Reply:
Noted about the paths, and thank you for the PR!